java/src/main/java/com/azure/graph/bulk/impl/model/GremlinVertex.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.graph.bulk.impl.model;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.util.HashMap;
+
+@Data
+@Builder
+@EqualsAndHashCode
+public class GremlinVertex {
+    private String id;
+    private String label;
+    private GremlinPartitionKey partitionKey;
+    private HashMap<String, GremlinVertexProperty> properties;
+
+    public void addProperty(String key, Object value) {
+        addProperty(key, value, false);
+    }
+
+    public void addProperty(String key, Object value, boolean isPartitionKey) {
+        if (value == null) return;
+
+        if (isPartitionKey) {
+            partitionKey = new GremlinPartitionKey(key, value.toString());
+        } else {
+            properties.put(key, new GremlinVertexProperty(value));
+        }
+    }
+
+    //TODO: expand the validation logic to include id and label
+    public void validate() {
```

Comment:
Please complete the validation -- these are important. We don't want customers to ingest a large amount of data in a format gremlin API can't read.